stubs/aiofiles/aiofiles/base.pyi

```diff
@@ -13,6 +13,9 @@ class AsyncBase(Generic[_T]):
     def __aiter__(self: Self) -> Self: ...
     async def __anext__(self) -> _T: ...
 
+class AsyncIndirectBase(AsyncBase[_T]):
+    def __init__(self, name: str, loop: Any, executor: Any, indirect: Callable[[], TextIO | BinaryIO]) -> None: ...
```

Comment:
I'm least confident about the type for indirect. In implementation indirect is only passed sys.stdin/sys.stdin.buffer/etc. It looks like intent is if _T is str then TextIO (or maybe IO[str]) while if _T is bytes then BinaryIO. 

Could add overloads to describe _T relationship although confused why _T is unconstrained type variable (seems like it should be str/bytes always).

Reply:
Hmm, I'd prefer to leave that for a follow up/future pr to be able to see effect by itself.

Reply:
sure